multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ProcessLogsPersistenceService.java

```diff
@@ -25,25 +25,47 @@
 import org.cloudfoundry.multiapps.controller.persistence.model.FileInfo;
 import org.cloudfoundry.multiapps.controller.persistence.model.ImmutableFileEntry;
 import org.cloudfoundry.multiapps.controller.persistence.model.ImmutableFileInfo;
+import org.cloudfoundry.multiapps.controller.persistence.model.OperationLogEntry;
 import org.cloudfoundry.multiapps.controller.persistence.query.providers.ByteArraySqlFileQueryProvider;
+import org.cloudfoundry.multiapps.controller.persistence.query.providers.SqlOperationLogQueryProvider;
 
 @Named("processLogsPersistenceService")
 public class ProcessLogsPersistenceService extends DatabaseFileService {
 
     public static final String TABLE_NAME = "process_log";
+    private final SqlOperationLogQueryProvider sqlOperationLogQueryProvider;
 
     public ProcessLogsPersistenceService(DataSourceWithDialect dataSourceWithDialect) {
         super(dataSourceWithDialect, new ByteArraySqlFileQueryProvider(TABLE_NAME, dataSourceWithDialect.getDataSourceDialect()));
+        sqlOperationLogQueryProvider = new SqlOperationLogQueryProvider();
     }
 
-    public List<String> getLogNames(String space, String operationId) throws FileStorageException {
+    public List<String> getLogNamesBackwardsCompatible(String space, String operationId) throws FileStorageException {
         List<FileEntry> logFiles = listFilesBySpaceAndOperationId(space, operationId);
         return logFiles.stream()
                        .map(FileEntry::getName)
                        .distinct()
                        .collect(Collectors.toList());
     }
 
+    public List<String> getLogNames(String space, String operationId) throws FileStorageException {
+        List<OperationLogEntry> operationLogEntries = listOperationLogsBySpaceAndOperationId(space, operationId);
+        return operationLogEntries.stream()
+                                  .map(OperationLogEntry::getOperationLogName)
+                                  .distinct()
+                                  .collect(Collectors.toList());
```

Comment:
if this distinct is only used here, can we add it in the query? will save some memory